offer-manager-domain/src/main/groovy/life/qbic/business/offers/content/CreateOfferContent.groovy

```diff
@@ -0,0 +1,176 @@
+        String creationDate = dateFormat.format(offer.modificationDate)
+        String expirationDate = offer.expirationDate.toLocalDate().toString()
+        groupProductItems(offer.items)
+
+        def offerContent = new OfferContent.Builder(offer.customer,offer.selectedCustomerAffiliation,offer.projectManager,creationDate,expirationDate,offer.projectTitle
+        ,offer.projectObjective, offer.experimentalDesign.orElse(""),offer.identifier.toString())
+
+        //collect productitems and convert to offeritems
+        groupProductItems(offer.items)
+        List<OfferItem> dataManagementOfferItems = dataManagementItems.collect{createOfferItem(it)}
+        List<OfferItem> dataAnalysisOfferItems = dataAnalysisItems.collect{createOfferItem(it)}
+        List<OfferItem> dataGenerationOfferItems = dataGenerationItems.collect{createOfferItem(it)}
+
+        offerContent.dataGenerationItems(dataGenerationOfferItems)
+        .dataAnalysisItems(dataAnalysisOfferItems)
+        .dataManagementItems(dataManagementOfferItems)
+
+        double overheadsDA = calculateOverheadSum(dataAnalysisOfferItems)
+        double overheadsDG = calculateOverheadSum(dataGenerationOfferItems)
+        double overheadsPMandDS = calculateOverheadSum(dataManagementOfferItems)
+
+        offerContent.overheadsDataAnalysis(overheadsDA)
+        .overheadsDataGeneration(overheadsDG)
+        .overheadsProjectManagementAndDataStorage(overheadsPMandDS)
+        .overheadTotal(offer.overheadSum)
+
+        offerContent.netDataAnalysis(calculateNetSum(dataAnalysisOfferItems))
+        .netDataGeneration(calculateNetSum(dataGenerationOfferItems))
+        .netProjectManagementAndDataStorage(calculateNetSum(dataManagementOfferItems))
+        .netCost(offer.totalNetPrice)
+
+        offerContent.totalVat(offer.taxCosts)
+        .vatRatio(offer.determineTaxCost())
+        .totalCost(offer.totalCosts)
+
+        OfferContent offerContentFinal = offerContent.build()
+
+        output.createdOfferContent(offerContentFinal)
+    }
+
+    /**
+     * Helper method that calculates the NET price for a list of product items
+     * @param offerItems The product item list for which the NET is calculated
+     *  @return The net value for the given list of items
+     */
+    private static double calculateNetSum(List<OfferItem> offerItems) {
+        double netSum = 0
+        offerItems.each {
+            netSum += it.quantity * it.unitPrice
+        }
+        return netSum
+    }
+
+    /**
+     * Calculates the overhead sum of all product items
+     * @param offerItems Items for which the overheads are calculated
+     * @return The overhead price for all items
+     */
+    private double calculateOverheadSum(List<OfferItem> offerItems) {
+        double overheadSum = 0.0
+        offerItems.each {
+            overheadSum += it.quantity * it.unitPrice * overheadRatio
+        }
+        return overheadSum
```

Comment:
```suggestion
        return offerItems.sum { it.quantity * it.unitPrice * overheadRatio }
```
I don't know whether you will run into errors here. Double values are not very precise. It might be fine for the time being.